packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx

```diff
@@ -0,0 +1,455 @@
+    },
+  ];
+
+  const tableRows = allOrders?.map((order, id) => ({
+    id: `${id}`,
+    startDate: {
+      sortKey: dayjs(order.dateActivated).toDate(),
+      content: (
+        <div className={styles.startDateColumn}>
+          <span>{formatDate(new Date(order.dateActivated))}</span>
+          {!isPrinting && <InfoTooltip orderer={order.orderer?.person?.display ?? '--'} />}
+        </div>
+      ),
+    },
+    orderId: order.orderNumber,
+    dateOfOrder: formatDate(new Date(order.dateActivated)),
+    orderType: capitalize(order.orderType?.display ?? '--'),
+    order: order.display,
+    priority: (
+      <div style={{ background: orderPriorityToColor(order.urgency), textAlign: 'center', borderRadius: '1rem' }}>
+        {capitalize(order.urgency)}
+      </div>
+    ),
+    orderedBy: order.orderer?.display,
+    status: capitalize(order.fulfillerStatus ?? '--'),
+    // actions: 'TBD'
+  }));
+
+  const sortRow = (cellA, cellB, { sortDirection, sortStates }) => {
+    return sortDirection === sortStates.DESC
+      ? compare(cellB.sortKey, cellA.sortKey)
+      : compare(cellA.sortKey, cellB.sortKey);
+  };
+
+  const sortDate = (myArray, order) =>
+    order === 'ASC'
+      ? orderBy(myArray, [(obj) => new Date(obj.encounterDate).getTime()], ['desc'])
+      : orderBy(myArray, [(obj) => new Date(obj.encounterDate).getTime()], ['asc']);
+
+  const { key, order } = sortParams;
+  const sortedData =
+    key === 'dateOfOrder'
+      ? sortDate(tableRows, order)
+      : order === 'DESC'
+      ? orderBy(tableRows, [key], ['desc'])
+      : orderBy(tableRows, [key], ['asc']);
+
+  const { results: paginatedOrders, goTo, currentPage } = usePagination(sortedData, defaultPageSize);
+
+  const patientDetails = useMemo(() => {
+    const getGender = (gender: string): string => {
+      switch (gender) {
+        case 'male':
+          return t('male', 'Male');
+        case 'female':
+          return t('female', 'Female');
+        case 'other':
+          return t('other', 'Other');
+        case 'unknown':
+          return t('unknown', 'Unknown');
+        default:
+          return gender;
+      }
+    };
+
+    const identifiers =
+      patient?.patient?.identifier?.filter(
+        (identifier) => !excludePatientIdentifierCodeTypes?.uuids.includes(identifier.type.coding[0].code),
+      ) ?? [];
+
+    return {
+      name: `${patient?.patient?.name?.[0]?.given?.join(' ')} ${patient?.patient?.name?.[0].family}`,
+      age: age(patient?.patient?.birthDate),
+      gender: getGender(patient?.patient?.gender),
+      location: patient?.patient?.address?.[0].city,
+      identifiers: identifiers?.length ? identifiers.map(({ value, type }) => value) : [],
+    };
+  }, [patient, t, excludePatientIdentifierCodeTypes?.uuids]);
+
+  // trigger a refetch to parent component if selectedOrderType changes
+  useEffect(() => {
+    // TODO: this is a hack to trigger a refetch of the parent component
+  }, [selectedOrderTypeUuid]);
```

Comment:
Can't we have the `selectedOrderTypeUuid` state be created in the parent itself?
Thanks!